pkgs/applications/office/kmymoney/default.nix

```diff
@@ -10,48 +17,36 @@
 # Needed for running tests:
 , qtbase, xvfb-run
 
-, python2, python3Packages
+, python3
 }:
 
-stdenv.mkDerivation rec {
+let
+  python = python3.withPackages (ps: with ps; [ weboob ]);
+in mkDerivation rec {
   pname = "kmymoney";
-  version = "5.1.1";
+  version = "5.1.2";
 
   src = fetchurl {
     url = "mirror://kde/stable/kmymoney/${version}/src/${pname}-${version}.tar.xz";
-    sha256 = "sha256-33ufeOhZb5nSgpXKc4cI8GVe4Fd4nf2SHHsbq5ZXgpg=";
+    hash = "sha256-N73E52OihJufc59z44s4nAK94cGxhE7c+n46sdW/ezs=";
   };
 
-  # Hidden dependency that wasn't included in CMakeLists.txt:
-  NIX_CFLAGS_COMPILE = "-I${kitemmodels.dev}/include/KF5";
-
   nativeBuildInputs = [
-    doxygen extra-cmake-modules graphviz kdoctools python2
-    python3Packages.wrapPython wrapQtAppsHook
+    doxygen extra-cmake-modules graphviz kdoctools python
+    wrapQtAppsHook
   ];
 
   buildInputs = [
     akonadi alkimia aqbanking gmp gwenhywfar kactivities karchive kcmutils
     kcontacts kdewebkit kdiagram kholidays kidentitymanagement kitemmodels
     libical libofx qgpgme
     sqlcipher
-
-    # Put it into buildInputs so that CMake can find it, even though we patch
-    # it into the interface later.
-    python3Packages.weboob
   ];
 
-  weboobPythonPath = [ python3Packages.weboob ];
-
-  postInstall = ''
-    buildPythonPath "$weboobPythonPath"
-    patchPythonScript "$out/share/kmymoney/weboob/kmymoneyweboob.py"
-
-    # Within the embedded Python interpreter, sys.argv is unavailable, so let's
-    # assign it to a dummy value so that the assignment of sys.argv[0] injected
-    # by patchPythonScript doesn't fail:
-    sed -i -e '1i import sys; sys.argv = [""]' \
-      "$out/share/kmymoney/weboob/kmymoneyweboob.py"
+  preFixup = ''
+    qtWrapperArgs+=(
+      --prefix PYTHONPATH : "${python}/${python.sitePackages}"
```

Comment:
This doesn't work; KMyMoney doesn't seem to respect `$PYTHONPATH`.

Reply:
Does someone know enough about the interactions between C++ and Python to find out how to set `$PYTHONPATH` in https://github.com/KDE/kmymoney/blob/v5.1.2/kmymoney/plugins/weboob/interface/weboobinterface.cpp?